git-subhistory.sh

```diff
@@ -351,7 +356,7 @@ cd ./$(git rev-parse --show-cdup) || exit $?
 GIT_DIR="$(git rev-parse --git-dir)"
 
 # a temporary directory for e.g. filter-branch backups
-mkdir "$GIT_DIR/subhistory-tmp/" || exit $?
+mkdir -p "$GIT_DIR/subhistory-tmp/" || exit $?
```

Comment:
what is the situation where the `.git` directory doesn't already exist that we want to fail silently?